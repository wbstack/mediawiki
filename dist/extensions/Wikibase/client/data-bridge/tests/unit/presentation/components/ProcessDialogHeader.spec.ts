import ProcessDialogHeader from '@/presentation/components/ProcessDialogHeader.vue';
import { shallowMount } from '@vue/test-utils';

describe( 'ProcessDialogHeader', () => {
	it( 'renders correctly without slots filled', () => {
		const wrapper = shallowMount( ProcessDialogHeader );
		expect( wrapper.element ).toMatchSnapshot();
	} );

	it( 'renders correctly with all props and slots filled', () => {
		const wrapper = shallowMount( ProcessDialogHeader, {
			slots: {
				title: '<span>title</span>',
				primaryAction: '<button>primary action</button>',
				safeAction: '<button>safe action</button>',
			},
		} );
		expect( wrapper.element ).toMatchSnapshot();
	} );

	it( 'gets content through the primaryAction slot', () => {
		const message = 'primary action';
		const wrapper = shallowMount( ProcessDialogHeader, {
			slots: { primaryAction: `<a class="mockPrimaryActionButton">${message}</a>` },
		} );
		expect( wrapper.find( 'a' ).text() ).toBe( message );
	} );

	it( 'gets content through the safeAction slot', () => {
		const message = 'safe action';
		const wrapper = shallowMount( ProcessDialogHeader, {
			slots: { safeAction: `<a class="mockSafeActionButton">${message}</a>` },
		} );
		expect( wrapper.find( 'a' ).text() ).toBe( message );
	} );

	it( 'gets title through the title slot', () => {
		const message = 'some message';
		const wrapper = shallowMount( ProcessDialogHeader, {
			slots: { title: `<span class="mockTitle">${message}</span>` },
		} );
		expect( wrapper.find( 'h1 span' ).text() ).toBe( message );
	} );

} );
