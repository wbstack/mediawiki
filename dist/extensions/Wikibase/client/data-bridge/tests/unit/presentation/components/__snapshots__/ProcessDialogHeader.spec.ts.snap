// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProcessDialogHeader renders correctly with all props and slots filled 1`] = `
<div
  class="wb-ui-processdialog-header"
>
  <!-- @slot This slot should be filled with a "safe" action, e.g. close -->
  
  <button>
    safe action
  </button>
  
  <h1
    class="wb-ui-processdialog-header__title"
  >
    <!-- @slot This slot should be filled with the title of the dialog -->
    
    <span>
      title
    </span>
    
  </h1>
  <!-- @slot This slot should be filled with the primary action, e.g. publish -->
  
  <button>
    primary action
  </button>
  
</div>
`;

exports[`ProcessDialogHeader renders correctly without slots filled 1`] = `
<div
  class="wb-ui-processdialog-header"
>
  <!-- @slot This slot should be filled with a "safe" action, e.g. close -->
  
  <div />
  
  <h1
    class="wb-ui-processdialog-header__title"
  >
    <!-- @slot This slot should be filled with the title of the dialog -->
    
    <span />
    
  </h1>
  <!-- @slot This slot should be filled with the primary action, e.g. publish -->
  
  <div />
  
</div>
`;
